### PR TITLE
improve: Return a copy of RefundRequestWithBlock objects

### PR DESCRIPTION
**Pull request by pxrl**
Removes a potential sharp edge if the recipient of this array happens to call Array.pop().

### CODE REVIEW

**Review by pxrl — COMMENTED** [review]
IMO we probably want to apply this to all of the `get...()` methods that return event data.

File: src/clients/SpokePoolClient.ts [hunk 1]
```diff
@@ -198,9 +198,10 @@ export class SpokePoolClient {
    * @returns A list of refund requests.
    */
   public getRefundRequests(fromBlock?: number, toBlock?: number): RefundRequestWithBlock[] {
-    return fromBlock === undefined || toBlock === undefined || isNaN(fromBlock) || isNaN(toBlock)
+    const refundRequests = fromBlock === undefined || toBlock === undefined || isNaN(fromBlock) || isNaN(toBlock)
       ? this.refundRequests
       : this.refundRequests.filter((request) => request.blockNumber >= fromBlock && request.blockNumber <= toBlock);
+    return [ ...refundRequests ];
```
Comment (james-a-morris) [inline src/clients/SpokePoolClient.ts:204]: If we're looking to deep-copy a variable, should we use `deepCopy`?
```suggestion
    return deepCopy(refundRequests);
```

Comment (james-a-morris) [inline src/clients/SpokePoolClient.ts:204]: Formatting per our linter file
```suggestion
    const refundRequests =
      fromBlock === undefined || toBlock === undefined || isNaN(fromBlock) || isNaN(toBlock)
        ? this.refundRequests
        : this.refundRequests.filter((request) => request.blockNumber >= fromBlock && request.blockNumber <= toBlock);
    return deepCopy(refundRequests);
```

**Review by james-a-morris — COMMENTED** [review]
LGTM - left a few nits

File: src/clients/SpokePoolClient.ts (hunk 1 above)
Comment (nicholaspai) [reply src/clients/SpokePoolClient.ts:204]: what's the difference? I think using native TS syntax might be better here

Comment (james-a-morris) [reply src/clients/SpokePoolClient.ts:204]: Sorry for the delay on this - I wasn't re-requested / notified of this. The `deepCopy` is going to deep copy the `refundRequest` elements whereas `[...refundRequests]` are just a shallow copy of the elements in a new array instance.

**Review by james-a-morris — COMMENTED** [review]
Left another comment

**Pull request closed without merging**